bam_sort.c

```diff
@@ -307,14 +307,15 @@ static void trans_tbl_init(bam_hdr_t* out, bam_hdr_t* translate, trans_tbl_t* tb
         regcomp(&rg_id_search, rg_regex.s, REG_EXTENDED|REG_NEWLINE|REG_NOSUB);
         free(rg_regex.s);
         kstring_t transformed_id = { 0, 0, NULL };
-        bool transformed_equals_match;
+        bool transformed_equals_match, not_found_in_output;
```

Comment:
I'm a bit unsure of how not_found_in_output is initialised.  It only gets set when rg_override is NULL.  Maybe we need a default value?

I couldn't see how to produce a test case where rg_override is NULL though so I don't know if this is possible.